packages/datepicker/src/react/use-key-events.ts

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { format, endOfWeek, startOfWeek, add, sub, getMonth } from 'date-fns'
+import { useDebounceCallback } from '@pluralsight/ps-design-system-util'
+
+interface UseKeyEvents {
+  selected: Date
+  months: number[]
+  handleForwardClick: React.MouseEventHandler<HTMLButtonElement>
+  handleBackClick: React.MouseEventHandler<HTMLButtonElement>
+}
+export const useKeyEvents = ({
+  selected,
+  months,
+  handleForwardClick,
+  handleBackClick
+}: UseKeyEvents) => {
+  const [focusedDate, setFocusedDate] = React.useState<Date | undefined>(
+    selected
+  )
+  const updateFocused = () => {
+    focusedDate &&
+      document
+        .querySelector<HTMLButtonElement>(
+          `[aria-label="${format(focusedDate, 'EEE LLL dd yyyy')}"]`
+        )
+        ?.focus()
+  }
+
+  const focusNext = useDebounceCallback(updateFocused, 100)
+
+  React.useEffect(() => {
+    focusNext()
+  }, [focusedDate, focusNext])
+
+  const dayKeyHandlers = (
+    date: Date,
+    dateProps: {
+      [x: string]: any
+    }
+  ) => {
+    const handleDayShift = (
+      dir: 1 | -1,
+      shift: 1 | 7,
+      evt: React.KeyboardEvent<HTMLButtonElement>
+    ) => {
+      const nextDate =
+        dir === 1
+          ? add(date, {
+              days: shift
+            })
+          : sub(date, {
+              days: shift
+            })
+      setFocusedDate(nextDate)
+      if (!months.includes(getMonth(nextDate))) {
+        dir === 1
+          ? handleForwardClick(
+              evt as unknown as React.MouseEvent<HTMLButtonElement>
```

Comment:
Not a huge deal based on today's implementation, I believe, but since a KeyboardEvent is not a MouseEvent and differs in its properties, it might not be great to pass one event to the other's handler.

I would suggest an event type-agnostic function like `moveForwardMonth` or something of that sort. What do you think?

Reply:
The library we're use only gives us this signature to use. It's an onClick I have to type cast like so to use this functionality to correctly move the month. I'm basing this off the libraries own code examples. It cannot be changed.